src/rpp/rpp/operators/flat_map.hpp

```diff
@@ -0,0 +1,68 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/map.hpp>
+#include <rpp/operators/merge.hpp>
+#include <rpp/defs.hpp>
+
+namespace rpp::operators::details
+{
+
+template<rpp::constraint::decayed_type Fn>
+struct flat_map_t 
+{
+    RPP_NO_UNIQUE_ADDRESS Fn m_fn;
+
+    template<rpp::constraint::observable TObservable>
+        requires std::invocable<Fn, rpp::utils::extract_observable_type_t<TObservable>>
```

Comment:
There is also invocation result must be observable

Reply:
It is already required [here](https://github.com/victimsnino/ReactivePlusPlus/pull/374/files#diff-a7a85d7aedddd82f15cd11012d2afd0e7fae88c55b3199b276f2500341bd1057R62) isn't it enough?

Reply:
No, check this comment

https://github.com/victimsnino/ReactivePlusPlus/pull/374/files#r1225415597

Reply:
done